app/views/partials/mediator-config.html

```diff
@@ -0,0 +1,141 @@
+        </div>
+      </div>
+
+      <div ng-if="def.type === 'number'">
+        <input class="form-control" type="number" ng-model="config[def.param][$index]">
+      </div>
+
+      <div ng-if="def.type === 'option'">
+        <select class="form-control" ng-options="val for val in def.values" ng-model="config[def.param][$index]"/>
+      </div>
+
+      <div ng-if="def.type === 'map'">
+        <table class="mediator-config-map">
+          <tr ng-repeat="(mapping, _v) in config[def.param][$index]">
+            <td>{{mapping}}</td>
+            <td><input class="form-control" type="text" ng-model="config[def.param][$parent.$index][mapping]"></td>
+            <td><button type="button" class="btn btn-danger btn-xs" ng-click="removeMappingInArray(def.param, $parent.$index, mapping)"><i class="glyphicon glyphicon-remove"/></button></td>
+          </tr>
+          <tr>
+            <td><input class="form-control" type="text" ng-model="inputKeysForArrays[def.param + $parent.$index]"></td>
+            <td><input class="form-control" type="text" ng-model="inputValuesForArrays[def.param + $parent.$index]"></td>
+            <td><button type="button" class="btn btn-primary btn-xs" ng-click="addNewMappingInArray(def.param, $parent.$index)" ng-disabled="!isNewKeyValidInArray(def.param, $parent.$index)"><i class="glyphicon glyphicon-plus"/></button></td>
+          </tr>
+          <tr ng-if="doesNewKeyExistInArray(def.param, $parent.$index)">
+            <td><div class="has-error"><i class="glyphicon glyphicon-warning-sign"></i>Key already exists</div></td>
+            <td></td>
+            <td></td>
+          </tr>
+        </table>
+      </div>
+
+      <div ng-if="def.type === 'struct'">
+        <div mediator-nested-config config-defs="def.template" config="config[def.param][$index]"></div>
+      </div>
+      </div><!-- col-sm-8 -->
+
+      <div class="col-sm-1 text-align-right">
+        <button type="button" class="btn btn-danger btn-xs" ng-click="removeArrayItem(def.param, $index)"><i class="glyphicon glyphicon-remove"/></button>
+      </div><!-- col-sm-1 -->
+    </div><!-- ng-repeat configItem in config[def.param] -->
+    <div>
+      <div class="col-sm-3">
+        <div ng-if="!config[def.param] || config[def.param].length == 0">
+          <label ng-if="def.displayName">{{def.displayName}}</label>
+          <label ng-if="!def.displayName">{{def.param}}</label>
+          <i ng-if="def.description" tooltip="{{def.description}}" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
+        </div>
+      </div><!-- col-sm-3 -->
+      <div class="col-sm-9 text-align-right">
+        <button type="button" class="btn btn-primary btn-xs" ng-click="addNewArrayItem(def)"><i class="glyphicon glyphicon-plus"/> <span ng-if="def.displayName">{{def.displayName}}</span><span ng-if="!def.displayName">{{def.param}}</span></button>
```

Comment:
I think we need just a litle bit of top margin here to space this away from the box:

![image](https://cloud.githubusercontent.com/assets/266576/11418701/097e034c-942b-11e5-80a6-bd8d199ce204.png)